### DISCUSSION TITLE
Some strange problems

### DISCUSSION THREAD

**Discussion by cworld1** [amannn/next-intl#1450] [Q&A] [answered]
1. The `router.replace(pathname, { locale: newLocale })` has some work, but locale is wrong.
2. If choose the same locale again, it crashed.
3. Half of components using intl's `<Link>`, has `undefined` locale render link.
4. I must add page `root` then redirect to make it work.
5. I cannot change the text on page loaded by `const t = useTranslations("HomePage")` when changing to a new locale setting.

![image](https://github.com/user-attachments/assets/d3ca45f2-2bec-4404-8ddc-50b67b4c228e)

[[Video removed]]

I have tried some ways to fix it, even copying some code of example. I don't know if there's any code have conflict to contentplayer2.

I have uploaded my repository on my account if anyone want to know more about the problem: https://github.com/cworld1/cqlink-test

**Comment by amannn** [comment, marked as answer]
Currently on my mobile, but one thing I noticed is that you’ve misspelled “middleware.ts”: https://github.com/cworld1/cqlink-test/blob/main/src/midware.ts

**Comment by cworld1** [comment]
@amannn Thank you! I don't know why I named it that name, but in my habit, i would like to copy/paste it instead of imputing it on my own.
